packages/common/src/util/Disposer.ts

```diff
@@ -0,0 +1,30 @@
+import { Disposable } from "../ide/types/ide.types";
+
+/**
+ * A class that can be used to dispose of multiple disposables at once. This is
+ * useful for managing the lifetime of multiple disposables that are created
+ * together. It ensures that if one of the disposables throws an error during
+ * disposal, the rest of the disposables will still be disposed.
+ */
+export class Disposer implements Disposable {
```

Comment:
see if we actually need to dynamically push anywhere. If not, just have a function

Reply:
@AndreasArvidsson looks like we do use dynamic push [here](https://github.com/cursorless-dev/cursorless/pull/1940/files#diff-404dfa5929061d4e42ab1fa85c49e3b31291731fc6aeaac6602fa48f618f9802R38-R42). So I'd be tempted to leave it as a class. Shall I leave it as `Disposer`? Maybe `CompositeDisposable`? `RobustCompositeDisposable` to signal it proceeds if there's an error in one of them?

Reply:
ok managed to remove all dynamic usage, and turns out we don't actually use any composites in this PR; changed to composite function and moved it to the next PR